examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,243 @@
+
+	// Start up TCP listener so the user can interact with the GenAI endpoint manually
+	if simConn == nil {
+		go tcpListener(*port, genAiConn)
+	} else {
+		log.Printf("Starting autonomous chat with Prompt: %s", *prompt)
+		go autonomousChat(*prompt, genAiConn, simConn, *numChats)
+	}
+
+	<-sigCtx.Done()
+	os.Exit(0)
+}
+
+func initClient(endpoint string, context string, name string) *connection {
+	// TODO: create option for a client certificate
+	client := &http.Client{}
+	conn := &connection{client: client, endpoint: endpoint, context: context, name: name}
+	return conn
+}
+
+type connection struct {
+	client   *http.Client
+	endpoint string // full base URL for API requests
+	context  string
+	name     string // human readable name for the connection
+	// TODO: create options for routes off the base URL
+}
+
+type GenAIRequest struct {
+	Context string `json:"context,omitempty"`
+	Prompt  string `json:"prompt"`
+}
+
+func handleGenAIRequest(prompt string, clientConn *connection) (string, error) {
+	jsonRequest := GenAIRequest{
+		Context: clientConn.context,
+		Prompt:  prompt,
+	}
+	jsonStr, err := json.Marshal(jsonRequest)
+	if err != nil {
+		return "", fmt.Errorf("unable to marshal json request: %v", err)
+	}
+
+	req, err := http.NewRequest("POST", clientConn.endpoint, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return "", fmt.Errorf("unable create http POST request: %v", err)
+	}
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := clientConn.client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("unable to post request: %v", err)
+	}
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("unable to read response body: %v", err)
+	}
+	defer resp.Body.Close()
+	body := string(responseBody)
+
+	if resp.StatusCode != 200 {
+		err = fmt.Errorf("Status: %s, Body: %s", resp.Status, body)
+	}
+	return string(responseBody) + "\n", err
+}
+
+// Two AIs (connection endpoints) talking to each other
+func autonomousChat(prompt string, conn1 *connection, conn2 *connection, numChats int) {
+	if numChats <= 0 {
+		return
+	}
+	response, err := handleGenAIRequest(prompt, conn1)
+	if err != nil {
+		log.Fatalf("could not send request: %v", err)
+	} else {
+		log.Printf("%d %s RESPONSE: %s\n", numChats, conn1.name, response)
+	}
+
+	numChats -= 1
+	// Flip between the connection that the response is sent to
+	autonomousChat(response, conn2, conn1, numChats)
+}
+
+// Manually interact via TCP with the GenAI endpont
+func tcpListener(port string, genAiConn *connection) {
+	log.Printf("Starting TCP server, listening on port %s", port)
+	ln, err := net.Listen("tcp", ":"+port)
+	if err != nil {
+		log.Fatalf("Could not start TCP server: %v", err)
+	}
+	defer ln.Close() // nolint: errcheck
+
+	for {
+		conn, err := ln.Accept()
+		if err != nil {
+			log.Printf("Unable to accept incoming TCP connection: %v", err)
```

Comment:
I think you need to add a `continue` after logging the error (or use if/else). Otherwise you still pass the connection to the `tcpHandleConnection` function.

Reply:
Ah, I meant to make this fatal.